File: server/internal/model/input/adminin/queue.go
package adminin

// QueueConfigSaveInp 保存队列 Topic 配置
type QueueConfigSaveInp struct {
	Id            uint64 `p:"id"            json:"id"            dc:"配置ID"`
	Topic         string `p:"topic"         json:"topic"         v:"required#Topic不能为空" dc:"Topic 标识"`
	Title         string `p:"title"         json:"title"         dc:"显示名称"`
	Workers       int    `p:"workers"       json:"workers"       v:"required|min:1|max:64#Worker数必填|Worker至少为1|Worker最多64" dc:"并行 Worker 数"`
	MaxRetry      int    `p:"maxRetry"      json:"maxRetry"      v:"min:0|max:20#重试次数0-20" d:"3" dc:"最大重试次数"`
	RetryDelaySec int    `p:"retryDelaySec" json:"retryDelaySec" v:"min:0|max:86400#重试间隔0-86400秒" d:"0" dc:"重试间隔秒"`
	Status        int    `p:"status"        json:"status"        v:"in:0,1#状态无效" d:"1" dc:"状态:0禁用,1启用"`
	Remark        string `p:"remark"        json:"remark"        dc:"备注"`
	Sort          int    `p:"sort"          json:"sort"          d:"0" dc:"排序"`
}

File: web/src/api/backend/system/queue.ts
/**
 * 消息队列 API
 */
import { adminRequest } from '@/utils/http'

export interface QueueTopicStats {
  topic: string
  title?: string
  pending: number
  deadSize: number
  rate?: number
  avgTakeMs?: number
  workers?: number
  maxRetry?: number
  retryDelaySec?: number
  status?: number
  configId?: number
  remark?: string
}

/** 获取队列统计 */
export function fetchQueueStats() {
  return adminRequest.get<{ driver: string; topics: QueueTopicStats[] }>({ url: '/queue/stats' })
}

/** 获取已注册 Topic 列表 */
export function fetchQueueTopics() {
  return adminRequest.get<{ list: string[] }>({ url: '/queue/topics' })
}

/** 测试投递消息（delaySec>0 为延迟投递） */
export function fetchQueuePushTest(params: { topic: string; body: string; delaySec?: number }) {
  return adminRequest.post({ url: '/queue/pushTest', params })
}

/** 保存 Topic 运行配置 */
export function fetchQueueConfigSave(params: {
  id?: number
  topic: string
  title?: string
  workers: number
  maxRetry?: number
  retryDelaySec?: number
  status?: number
  remark?: string
  sort?: number
}) {
  return adminRequest.post({ url: '/queue/configSave', params })
}
